streaming/src/main/scala/org/apache/spark/streaming/receiver/ActorReceiver.scala

```diff
@@ -149,43 +181,61 @@ private[streaming] class ActorReceiver[T: ClassTag](
   class Supervisor extends Actor {
 
     override val supervisorStrategy = receiverSupervisorStrategy
-    val worker = context.actorOf(props, name)
-    logInfo("Started receiver worker at:" + worker.path)
-
-    val n: AtomicInteger = new AtomicInteger(0)
-    val hiccups: AtomicInteger = new AtomicInteger(0)
-
```

Comment:
Hi, @sarutak , I went back to Akka's document http://doc.akka.io/docs/akka/snapshot/general/jmm.html (Actors and the Java Memory Model), I think they stated that, 

> "internal fields of the actor are visible when the next message is processed by that actor. So fields in your actor need not be volatile or equivalent. "

So, we don't need to explicitly mark these variables to be volatile?